doc/topics/releases/2015.8.4.rst

```diff
@@ -0,0 +1,15 @@
+===========================
+Salt 2015.8.4 Release Notes
+===========================
+
+Core Changes
+============
+
+- Added ``state.validate`` feature.  This checks SLS data against the salt
+  state API and reports errors if the argument types do not match the types
```

Comment:
Agreed. The docs will need to be quite specific about what it does,
when, and why.

`state.validate` will improve incrementally in the same vein as
`predictive_changes` since we must manually make the necessary
additions to each state function. In this case we will need to add
Sphinx annotations to the existing argument docs that define things
like "int or list".